tests/Unit/Utilities/PregTest.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Tests\Unit\Utilities;
+
+use Pelago\Emogrifier\Utilities\Preg;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Pelago\Emogrifier\Utilities\Preg
+ */
+final class PregTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function throwExceptionsProvidesFluentInterface(): void
+    {
+        $subject = new Preg();
+
+        $result = $subject->throwExceptions(false);
+
+        self::assertSame($subject, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function exceptionNotThrownByDefault(): void
+    {
+        $this->expectNotToPerformAssertions();
+        $subject = new Preg();
+
+        // Version 6 of PHPUnit allowed setting `PHPUnit\Framework\Error\Notice::$enabled` to `false`
+        // to prevent its error handler throwing exceptions upon `trigger_error`.
+        // This functionality has been removed by version 9.
+        // Version 10 allows use of the #[WithoutErrorHandler] attribute, but this requires PHP 8.0 anyway.
+        // So we must `@` to suppress the triggered error.
+        // This also means we can't test for our own triggered error, since `preg_*` also triggers an error.
+        @$subject->replace('/', '', '');
+    }
+
+    /**
+     * @test
+     */
+    public function exceptionThrownAfterThrowExceptionsTurnedOn(): void
+    {
+        $this->expectException(\RuntimeException::class);
+        $subject = new Preg();
+
+        $subject->throwExceptions(true);
+        @$subject->replace('/', '', '');
+    }
+
+    /**
+     * @test
+     */
+    public function exceptionNotThrownAfterThrowExceptionsTurnedOff(): void
+    {
+        $this->expectNotToPerformAssertions();
+        $subject = new Preg();
+
+        $subject->throwExceptions(true);
+        $subject->throwExceptions(false);
+        @$subject->replace('/', '', '');
+    }
+
+    /**
+     * @return array<array<array-key, string|non-empty-array<string>|int>>
```

Comment:
```suggestion
     * @return array<non-empty-string, array{pattern: string|list<non-empty-string>, replacement: non-empty-string|list<non-empty-string>, subject: non-empty-string , limit: int, expect: non-empty-string}>
```

Reply:
I did consider this, but found that the array keys have to be explicitly stated in the type, whereas we only care about the types of the values, in order.  However, I'm fine either way so have changed it (which does mean the method must now use the specific key names).